test/string-regex_test.go

```diff
@@ -0,0 +1,21 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/mgechev/revive/lint"
+	"github.com/mgechev/revive/rule"
+)
+
+func TestStringRegex(t *testing.T) {
+	testRule(t, "string-regex", &rule.StringRegexRule{}, &lint.RuleConfig{
+		Arguments: []interface{}{
+			[]interface{}{
+				"stringRegexMethod1", // The first argument is checked by default
+				"/^[A-Z]/",
+				"must start with a capital letter"},
+
+			[]interface{}{
+				"stringRegexMethod2[2].d",
```

Comment:
the ability to refer fields of structs might be "too powerful" and maybe not sufficiently useful for the complexity its implementation requires

Reply:
One of the main reasons I started working on this feature was because I knew it would be helpful for personal projects I'm working on, as well as hopefully others that do similar work. In an API I've been developing, [I use a struct to describe error responses](https://github.com/very-amused/CSplan-API/blob/2f0b07a6d4d56e8423bfe561e8007e29131fb1ce/core/errors.go#L13), which are then shown to users by a client, so it's desirable to be able to lint `core.WriteError[1].Message`.

I believe that this is a fairly common scenario, but it does have a tradeoff in terms of complexity, which is why I've limited the feature to top level fields.